QuizApp/App/Common/SceneDelegate.swift

```diff
@@ -36,9 +37,29 @@ private extension SceneDelegate {
         self.window = window
         let navigationController = UINavigationController()
         coordinator = Coordinator(navigationController: navigationController)
+        serviceFactory = ServiceFactory()
+        checkUserToken()
         window.rootViewController = navigationController
         window.makeKeyAndVisible()
-        coordinator.showLogin()
+
+    }
+
+    func checkUserToken() {
```

Comment:
This is all ok, but for a final touch, you should delete token if validation fails.